pkg/cloud/cloud_io.go

```diff
@@ -159,6 +194,7 @@ type ReaderOpenerAt func(ctx context.Context, pos int64) (io.ReadCloser, error)
 type ResumingReader struct {
 	Opener       ReaderOpenerAt   // Get additional content
 	Reader       io.ReadCloser    // Currently opened reader
+	Filename     string           // Used for logging
```

Comment:
If we don't think these logging changes are worth it, I can pull them out as well.